pom.xml

```diff
@@ -467,7 +467,7 @@
           <dependency>
             <groupId>com.puppycrawl.tools</groupId>
             <artifactId>checkstyle</artifactId>
-            <version>7.8</version>
+            <version>7.9-SNAPSHOT</version>
```

Comment:
I didn't think maven would work with dependency that defines itself if said dependency hasn't been installed yet as it gets all dependencies from the maven cache folder.
Can someone confirm it isn't using the previously installed SNAPSHOT version?
If this works as intended than https://github.com/checkstyle/checkstyle/pull/4403 isn't needed.